near-sdk/src/json_types/public_key.rs

```diff
@@ -1,19 +1,29 @@
-use borsh::{BorshDeserialize, BorshSerialize};
+use borsh::{maybestd::io, BorshDeserialize, BorshSerialize};
 use bs58::decode::Error as B58Error;
-use serde::{Deserialize, Serialize};
 use std::convert::TryFrom;
 
 /// PublicKey curve
-#[derive(
-    Debug, Clone, Copy, Serialize, Deserialize, PartialEq, BorshDeserialize, BorshSerialize,
-)]
+#[derive(Debug, Clone, Copy, PartialOrd, Ord, Eq, PartialEq, BorshDeserialize, BorshSerialize)]
+#[repr(u8)]
 pub enum CurveType {
     ED25519 = 0,
     SECP256K1 = 1,
 }
 
+impl TryFrom<u8> for CurveType {
+    type Error = ParsePublicKeyError;
+
+    fn try_from(value: u8) -> Result<Self, Self::Error> {
+        match value {
+            0 => Ok(Self::ED25519),
+            1 => Ok(Self::SECP256K1),
+            _ => Err(ParsePublicKeyError { kind: ParsePublicKeyErrorKind::UnknownCurve }),
+        }
+    }
+}
+
```

Comment:
This is a bit of an odd *public* API, getting a structured error here which also describes all other possible parse errors feels strange. Let's not add this impl, and instead add a private inherent `from_u8()` method (or just inline at the call-site).